Neos.ContentGraph.DoctrineDbalAdapter/src/ContentRepositoryReadModelAdapter.php

```diff
@@ -136,11 +142,23 @@ public function findContentStreams(): ContentStreams
      */
     private static function workspaceFromDatabaseRow(array $row): Workspace
     {
+        $baseWorkspaceName = $row['baseWorkspaceName'] !== null ? WorkspaceName::fromString($row['baseWorkspaceName']) : null;
+
+        $status = WorkspaceStatus::UP_TO_DATE;
+        if ($baseWorkspaceName !== null) {
+            // root workspaces can never be out of date and dont have a source content stream
+            $sourceVersion = MaybeVersion::fromVersionOrNull(Version::fromInteger($row['sourceVersion']));
+            $expectedSourceVersion = ExpectedVersion::fromVersion(Version::fromInteger($row['expectedSourceVersion']));
+
+            if (!$expectedSourceVersion->isSatisfiedBy($sourceVersion)) {
+                $status = WorkspaceStatus::OUTDATED;
+            }
```

Comment:
Mostly a naming thing, but the "expected version" is something for the write side, i.e. I trigger a command and I expect the model to be of a given version.
In this case it is just two versions of content streams.